Dalamud/Game/Network/Internal/NetworkHandlers.cs

```diff
@@ -376,85 +492,46 @@ private IDisposable HandleMarketBoardPurchaseHandler()
-                           using var stream = new UnmanagedMemoryStream((byte*)message.Data.ToPointer(), 64);
-                           using var reader = new BinaryReader(stream);
-
-                           var notifyType = reader.ReadByte();
-                           stream.Position += 0x1B;
-                           var conditionId = reader.ReadUInt16();
-
-                           if (notifyType != 3)
-                               return;
-
-                           var cfConditionSheet = message.DataManager!.GetExcelSheet<ContentFinderCondition>()!;
-                           var cfCondition = cfConditionSheet.GetRow(conditionId);
-
-                           if (cfCondition == null)
-                           {
-                               Log.Error("CFC key {ConditionId} not in Lumina data", conditionId);
-                               return;
-                           }
-
-                           var cfcName = cfCondition.Name.ToString();
-                           if (cfcName.IsNullOrEmpty())
-                           {
-                               cfcName = "Duty Roulette";
-                               cfCondition.Image = 112324;
-                           }
-
-                           // Flash window
-                           if (this.configuration.DutyFinderTaskbarFlash && !NativeFunctions.ApplicationIsActivated())
-                           {
-                               var flashInfo = new NativeFunctions.FlashWindowInfo
-                               {
-                                   Size = (uint)Marshal.SizeOf<NativeFunctions.FlashWindowInfo>(),
-                                   Count = uint.MaxValue,
-                                   Timeout = 0,
-                                   Flags = NativeFunctions.FlashWindow.All | NativeFunctions.FlashWindow.TimerNoFG,
-                                   Hwnd = Process.GetCurrentProcess().MainWindowHandle,
-                               };
-                               NativeFunctions.FlashWindowEx(ref flashInfo);
-                           }
-
-                           Task.Run(() =>
-                           {
-                               if (this.configuration.DutyFinderChatMessage)
-                               {
-                                   Service<ChatGui>.GetNullable()?.Print($"Duty pop: {cfcName}");
-                               }
-
-                               this.CfPop.InvokeSafely(cfCondition);
-                           }).ContinueWith(
-                               task => Log.Error(task.Exception, "CfPop.Invoke failed"),
-                               TaskContinuationOptions.OnlyOnFaulted);
-                       },
-                       ex => Log.Error(ex, "Failed to handle Market Board purchase event"));
+        return this.configuration.IsMbCollect;
     }
 
-    private bool ShouldUpload<T>(T any)
+    private nint MarketPurchasePacketDetour(nint a1, nint packetData)
```

Comment:
All of these also should have try/finally

Reply:
I don't see the need for try/finally here as it's already handled with the `?` for the invokes